### DISCUSSION TITLE
New parameter counterPartyId in EDC version 0.7.3

### DISCUSSION THREAD

**Discussion by ma3u** [eclipse-tractusx/tractusx-edc#1545] [Show and tell]
Our support engineers faced many faulty EDC catalog requests with deprecated V2 with missing parameter **counterPartyId**.

Please ensure the EDC version 0.7.3 with following parameter:

- `"counterPartyId": "providerId",`

You will get the error message from Identity Wallet:
- `[{"message":"Unable to obtain credentials: Empty optional","type":"BadGateway","path":null,"invalidValue":null}] `

The EDC v2 should fail with a error message.

[EDC version 0.7.4 Swagger API]( https://app.swaggerhub.com/apis/eclipse-tractusx-bot/tractusx-edc/0.7.4#/Catalog%20V3/requestCatalogV3) (0.7.3 swagger doesn't consist catalog API)

![image](https://github.com/user-attachments/assets/08669747-14f2-4f1f-933f-093d2f1f9425)

Swagger API for 0.7.4 isn't valid: in Line 486 _Operations must have unique operationIds._

 - `operationId: getTransferState`

**Comment by ma3u** [comment]
generated [EDC version 0.7.3](https://app.swaggerhub.com/apis/eclipse-tractusx-bot/tractusx-edc/0.7.3) doesn't consist the Catalog API: 

![image](https://github.com/user-attachments/assets/e0bc62bf-32ca-42ae-b433-de8c2b124c38)

**Comment by ma3u** [comment]
@lgblaumeiser can you take a look?

**Comment by lgblaumeiser** [comment]
@ma3u @pdrd

The property counterPartyId exists since 0.7.0. With version 0.7.3, the catalog api v3 has been added which results that v2 is deprecated. The corresponding issue as mentioned in the release notes of the upstream EDC version 0.7.1 is https://github.com/eclipse-edc/Connector/pull/4211. It contains the deprecation declaration of the v2 apis.

So looking at what you describe, the remaining issue is, that the v2 version of the catalog request does not work anymore due to a protocol mismatch with the wallet. But v3 does work properly, right? Anything, I overlooked?
